compiler/rustc_privacy/src/lib.rs

```diff
@@ -194,6 +194,16 @@ where
     }
 }
 
+struct LazyDefPathStr<'tcx> {
+    def_id: DefId,
+    tcx: TyCtxt<'tcx>,
+}
+impl<'tcx> fmt::Display for LazyDefPathStr<'tcx> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.tcx.def_path_str(self.def_id))
+    }
+}
```

Comment:
Nit: could you move this above `trait DefIdVisitor`? (And also format with an empty line between items.)
Right now it's copypasted right between two impls of `DefIdVisitorSkeleton`.